qiskit_nature/algorithms/initial_points/hf_initial_point.py

```diff
@@ -0,0 +1,160 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The HFInitialPoint class to compute an initial point for the VQE Ansatz parameters."""
+
+from __future__ import annotations
+
+import warnings
+
+import numpy as np
+
+from qiskit_nature.circuit.library import UCC
+from qiskit_nature.properties.second_quantization.electronic import ElectronicEnergy
+from qiskit_nature.properties.second_quantization.second_quantized_property import (
+    GroupedSecondQuantizedProperty,
+)
+from qiskit_nature.exceptions import QiskitNatureError
+
+from .initial_point import InitialPoint
+
+
+class HFInitialPoint(InitialPoint):
+    r"""Compute the Hartree-Fock (HF) initial point.
+
+    A class that provides an all-zero initial point for the ``VQE`` parameter values.
+
+    If used in concert with the
+    :class:`~qiskit.circuit.library.initial_states.hartree_fock.HartreeFock` initial state (which
+    will be prepended to the :class:`~qiskit.circuit.library.ansatzes.ucc.UCC` circuit) the all-zero
+    initial point will correspond to the HF initial point.
+
+    The excitation list generated by the :class:`~qiskit.circuit.library.ansatzes.ucc.UCC` ansatz is
+    obtained to ensure that the shape of the initial point is appropriate.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._ansatz: UCC | None = None
+        self._excitation_list: list[tuple[tuple[int, ...], tuple[int, ...]]] | None = None
+        self._grouped_property: GroupedSecondQuantizedProperty | None = None
+        self._reference_energy: float = 0.0
+        self._parameters: np.ndarray | None = None
+
+    @property
+    def grouped_property(self) -> GroupedSecondQuantizedProperty | None:
```

Comment:
Just to be even more explicit: in order to unify the electronic and vibrational structure cases,
I suggest that these attributes become part of the base class.

Reply:
Incorporated this change as part of [8c5b10f](https://github.com/Qiskit/qiskit-nature/pull/632/commits/8c5b10fc437a19c4020ce2b57d34afd5966beb3c).